lib/github-pages-health-check/domain.rb

```diff
@@ -170,7 +174,11 @@ def dns
         return unless valid_domain?
         @dns = Timeout.timeout(TIMEOUT) do
           GitHubPages::HealthCheck.without_warnings do
-            Net::DNS::Resolver.start(absolute_domain).answer unless host.nil?
+            unless host.nil?
+              resolver = Net::DNS::Resolver.new
+              resolver.search(absolute_domain, Net::DNS::A).answer +
+                resolver.search(absolute_domain, Net::DNS::MX).answer
```

Comment:
I added it here because otherwise we'd have to duplicate the timeouts, guards, and warning checks. 95% of the method would be the same. Maybe a third method which wraps around `Resolver#search` and allows you to specify the type you're interested in.